hca/parser.py

```diff
@@ -15,36 +23,331 @@ def make_name(http_method, path_split):
+        elif "properties" not in param:
+            return []
+
+        # If there are properties or it's within an array, loop through properties and add each.
+        # Will be an empty list if it's an array there are no defined properties. In this case,
+        # we should parse the input as json.
+        object_indexed_params = []
+        for (prop_name, payload) in param['properties'].items():
+            payload["name"] = prop_name
+            payload['req'] = param['req'] and prop_name in param["required"]
+            payload["in"] = param["in"]
+            object_indexed_params.append(_recursive_indexing(spec, payload, hierarchy_clone, indexed_parameters, arr))
+        return object_indexed_params
+
+    # If this within array, don't set the parameter because we set a parameter value in the array "if"
+    elif arr:
+        return param
+
+    # Normal type - add it to indexed parameters.
+    else:
+        param_name = param['name']
+        param['name'] = param_name
+        param["array"] = arr
+        param["hierarchy"] = hierarchy_clone
+        indexed_parameters[param_name] = param
+
+
+def index_parameters(spec, endpoint_info):
     """Map all parameters from parameter name -> parameter info."""
     indexed_parameters = {}
+    if "parameters" in endpoint_info:
+        for param in endpoint_info['parameters']:
 
-    if "parameters" not in endpoint_info:
-        return indexed_parameters
-
-    for param in endpoint_info['parameters']:
-        if "schema" in param:
-            schema_params = param['schema']['properties']
-            for schema_param_name in schema_params:
-                param_name = param['name'] + "-" + schema_param_name
-                payload = schema_params[schema_param_name]
-                payload["name"] = param_name
-                payload['required'] = False  # Placeholder before checking required array
-                payload['in'] = param["in"]
-                indexed_parameters[param_name] = payload
-            for required_name in param['schema']['required']:
-                param_name = param['name'] + "-" + required_name
-                indexed_parameters[param_name]['required'] = True
-        else:
-            param_name = param['name']
-            indexed_parameters[param_name] = param
+            # Labelling req instead of required b/c objects have "required"
+            # array to tell which args required. Can't overwrite.
+            param["req"] = param["required"]
+
+            # Make sure parameter name is not extras.
+            if param["in"] == "body":
+                inner_param = param["schema"]
+                inner_param["in"] = "body"
+                inner_param["name"] = "irrelevant"
+                inner_param['req'] = param['req']
+                param = inner_param
+
+            _recursive_indexing(spec, param, [], indexed_parameters)
     return indexed_parameters
 
 
+def create_help_statement(endpoint_info):
```

Comment:
Removing unhelpful message saying what arguments are required for which endpoints. In future pr, will throw an error if subcommand given without proper query string args. For instance, if GET bundles/:uuid/:version doesn't provide a replica argument.